core/commands/configure_network_command.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require_relative '../models/network_config_file'
+require_relative '../services/machine_configurator'
+require 'net/ssh'
+
+# This class loads ssh keys to configuration or selected nodes.
+class ConfigureNetworkCommand < BaseCommand
+  # Create the command instance.
+  # @param args [String] string of arguments splits / for the current command.
+  # @param keyFile [String] path for ssh key on local machine
+  # @param labels [String] label of nodes for the current command
+  # @param out [Out] the object that should be used to log information.
+  def initialize(args, env, out)
+    raise 'Configuration name is required' if args.nil?
+
+    @config = Configuration.new(args, env.labels)
+    @keyfile = env.keyFile
+    @out = out
+    @configfile = NetworkConfigFile.new(@config.network_settings_file)
+  end
+
+  # This method is called whenever the command is executed
+  def execute
+    public_keys
+  end
+
+  private
+
+  # copy ssh keys to config
+  def public_keys
+    exit_code = 0
+    nodes = @config.node_configurations
+    raise "No aws, vbox, libvirt, docker nodes found in #{@args[0]}" if nodes.empty?
```

Comment:
There is no point in rasing exception here. Please simply return the error code.